src/dmd/dmangle.d

```diff
@@ -111,6 +111,31 @@ unittest
     assert(!'\\'.isValidMangling);
 }
 
+/**********************************************
+ * Convert a string representing a type (the deco) and
+ * return its equivalent Type.
+ * Params:
+ *      deco = string containing the deco
+ * Returns:
+ *      null for failed to convert
+ *      Type for succeeded
+ */
+
+public Type decoToType(const(char)[] deco)
```

Comment:
Why not move this function to `traits.d` ? Then it can be made private.

Reply:
Because traits.d is plenty large enough, and it would be best suited to be with the mangling code.